src/server/scripts/Zandalar/Underrot/boss_cragmaw_the_infested.cpp

```diff
@@ -0,0 +1,377 @@
+
+    // Blood Tick
+    SPELL_BLOOD_BURST_DAMAGE            = 278637,
+    SPELL_SERRATED_FANGS                = 260455,
+
+    // Fetid Maggot
+    SPELL_FEIGN_DEATH                   = 159474
+};
+
+enum CragmawEvents
+{
+    // Cragmaw
+    EVENT_CHARGE_SELECTOR       = 1,
+    EVENT_INDIGESTION,
+    EVENT_NORMAL_REQUEUE,
+
+    // Blood Tick
+    EVENT_SERRATED_FANGS        = 1
+};
+
+enum Points
+{
+    POINT_TANTRUM_START_RND_MOVEMENT    = 0
+};
+
+enum CragmawNPC
+{
+    NPC_FETID_MAGGOT    = 130909
+};
+
+Position const FetidMaggotSpawn = { 857.864f, 984.981f, 39.231f, 4.68147f };
+
+// 131817 - Cragmaw the Infested
+struct boss_cragmaw_the_infested : public BossAI
+{
+    boss_cragmaw_the_infested(Creature* creature) : BossAI(creature, DATA_CRAGMAW_THE_INFESTED) { }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+    }
+
+    void JustAppeared() override
+    {
+        if (instance->GetData(DATA_CRAGMAW_CRAWG_EATING))
+            return;
+
+        DoCastSelf(SPELL_CRAWG_EATING);
+        me->SetEmoteState(EMOTE_STATE_EAT);
+        if (TempSummon* summon = me->SummonCreature(NPC_FETID_MAGGOT, FetidMaggotSpawn))
+        {
+            _fetidMaggotGuid = summon->GetGUID();
+            summon->CastSpell(nullptr, SPELL_FEIGN_DEATH, true);
+            summon->SetUnitFlag(UnitFlags(UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_IMMUNE_TO_NPC | UNIT_FLAG_UNINTERACTIBLE));
+        }
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        summons.DespawnAll();
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void ScheduleSpells()
+    {
+        events.Reset();
+
+        if (GetDifficulty() == DIFFICULTY_NORMAL)
+            events.ScheduleEvent(EVENT_NORMAL_REQUEUE, 45s);
+
+        if (urand(0, 1) == 0)
+        {
+            events.ScheduleEvent(EVENT_CHARGE_SELECTOR, 9s);
+            events.ScheduleEvent(EVENT_INDIGESTION, 20s);
+            events.ScheduleEvent(EVENT_CHARGE_SELECTOR, 33s);
+        }
+        else
+        {
+            events.ScheduleEvent(EVENT_INDIGESTION, 9s);
+            events.ScheduleEvent(EVENT_CHARGE_SELECTOR, 21s);
+            events.ScheduleEvent(EVENT_CHARGE_SELECTOR, 41s);
+        }
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 1);
+        instance->SetData(DATA_CRAGMAW_CRAWG_EATING, 1);
+        ScheduleSpells();
+
+        if (Creature* fetidMaggot = ObjectAccessor::GetCreature(*me, _fetidMaggotGuid))
+            fetidMaggot->DespawnOrUnsummon();
+
+        if (IsHeroic() || GetDifficulty() == DIFFICULTY_MYTHIC || GetDifficulty() == DIFFICULTY_MYTHIC_KEYSTONE)
+        {
+            me->SetPowerType(POWER_ENERGY);
+            DoCast(SPELL_TANTRUM_ENERGY_BAR);
```

Comment:
Same thing here, SPELL_AURA_MOD_OVERRIDE_POWER_DISPLAY is always set on JustAppeared. The only thing called in JustEngagedWith is the energizing aura.